citar-latex.el

```diff
@@ -0,0 +1,127 @@
+;;
+;; This file is not part of GNU Emacs.
+;;
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+;;
+;;; Commentary:
+;;
+;; A small package that provides the functions required to use citar
+;; with latex.
+;;
+;; Simply loading this file will enable manipulating the citations with
+;; commands provided by citar.
+;;
+;;; Code:
+
+(require 'citar)
+(require 'tex)
+(require 'reftex-parse)
+
+(defvar citar-major-mode-functions)
+
+(defcustom citar-latex-cite-commands
+  '((("cite" "Cite" "citet" "Citet" "parencite"
+      "Parencite" "footcite" "footcitetext" "textcite" "Textcite"
+      "smartcite" "Smartcite" "cite*" "parencite*" "autocite"
+      "Autocite" "autocite*" "Autocite*" "citeauthor" "Citeauthor"
+      "citeauthor*" "Citeauthor*" "citetitle" "citetitle*" "citeyear"
+      "citeyear*" "citedate" "citedate*" "citeurl" "fullcite"
+      "footfullcite" "notecite" "Notecite" "pnotecite" "Pnotecite"
+      "fnotecite") . (["Prenote"] ["Postnote"] t))
+    (("nocite" "supercite") . nil))
+  "Citation commands and their argument specs.
+
+The argument spec is the same as the args argument of
+`TeX-parse-macro'. When calling `citar-insert-citation' the keys
+will be inserted at the position where `TeX-parse-macro' leaves
+the point."
+  :group 'citar-latex
+  :type '(alist :key-type (repeat string)
+                :value-type sexp))
+
+(defcustom citar-latex-prompt-for-extra-arguments t
+  "Whether to prompt for additional arguments when inserting a citation."
+  :group 'citar-latex
+  :type 'boolean)
+
+(add-to-list 'citar-major-mode-functions
+             '((latex-mode) .
+               ((local-bib-files . citar-latex--local-bib-files)
+                (insert-keys . citar-latex--insert-keys)
+                (insert-citations . citar-latex--insert-citations)
+                (keys-at-point . citar-latex--keys-at-point))))
+
+(defun citar-latex--local-bib-files ()
+  "Retrieve local bibliographic files for a latex buffer using reftex."
+  (reftex-locate-bibliography-files (TeX-master-directory)))
+
+(defun citar-latex--keys-at-point ()
+  "Return a list of keys at point in a latex buffer."
+    (when (citar-latex-is-a-cite-command (TeX-current-macro))
+      (split-string (thing-at-point 'list t) "," t "[{} ]+")))
+
+(defun citar-latex--insert-keys (keys)
+  "Insert comma sperated KEYS in a latex buffer."
+  (insert (string-join keys ", ")))
```

Comment:
I don't believe this works for biblatex, which has this annoying (from our POV) distinction between two types of citations. Examples:

```latex
Test 1: \autocite{low2001,mcneill2011}
Test 2: \autocites{low2001}[see][p. 12]{mcneill2011}
```

Perhaps we just follow bibtex-completion for initial merge and only support for the first variant?

I still think we should have a separate command list for natbib and biblatex, and rely on `reftex-using-biblatex-p` to know which to use?